content/doc/developer/internationalization/new-internationalization.adoc

```diff
@@ -0,0 +1,339 @@
+
+Developers place messages that require localizations in
+`+Messages.properties+`. Those need to be translated in the usual
+manner. See `+Messages_ja.properties+` in the core as an example if you
+are new to this process.
+
+Sometimes looking at `+Messages.properties+` alone doesn't give you
+enough contextual information as to where the messages are used. For
+this, developers are encouraged to access messages by using the
+type-safe `+Messages+` class generated by
+http://java.net/projects/localizer/[localizer]. To find out where
+messages are actually used, use your IDE to find all the usages of the
+message format method.
+
+=== Translating message references in Jelly
+
+The other messages that need to be translated are in Jelly view files,
+which are in `+src/main/resources/**.jelly+`. To localize them, first
+you run Maven to generate skeleton property file for your locale:
+
+[width="100%",cols="100%",]
+|===
+a|
+....
+$ cd jenkins/core  (or a plugin dir)
+$ mvn stapler:i18n -Dlocale=fr
+....
+
+|===
+
+This will generate a bunch of `+*_fr.properties+` all over
+`+src/main/resources+` with an empty value. If the file already exists,
+it will append missing entries to existing files.
+
+You then need to work on each such property file and translate messages.
+You don't have to translate the entire file — if you leave some entries
+empty, they'll fall back to the default locale.
+
+[cols="",]
+|===
+|https://addons.mozilla.org/en-US/firefox/addon/1333[Quick locale
+switcher] firefox extension is useful to toggle between various locales.
+|===
+
+=== Translating static HTML resources
+
+Stand-alone HTML files are often used in Jenkins for things like inline
+help messages. These resources need to be translated by adding the
+locale code between the file name and the extension. For example, the
+Japanese version of `+abc.html+` would be `+abc_ja.html+`, and British
+version of it could be `+abc_en_GB.html+`. These files need to be
+encoded in UTF-8.
+
+=== Pushing changes
+
+Once you made some changes, you can commit them. Translators should
+consider themselves as owning property files for their locale, so feel
+free to go ahead and just commit. If you are new to this, doing a small
+commit first is a good idea. You can also always send in a patch if you
+prefer to be safe.
+
+When starting a translation, try to check if there's anyone else working
+on the same locale. You can find out who they are by finding existing
+localization and looking at its history. Try to get in touch with them
+to avoid a surprise.
+
+== Stapler plugin for IntelliJ IDEA
+
+Refactoring the existing code to handle i18n correctly is tedious. So
+https://wiki.jenkins.io/display/JENKINS/IntelliJ+IDEA+plugin+for+Stapler[IntelliJ
+IDEA plugin for Stapler] is developed to simplify this (note that
+JetBrains kindly offered the open-source license for the Jenkins
+project, so contributers can get the license for free — contact Kohsuke
+if you need one.)
```

Comment:
What is the proper way to obtain this license now?  If it really is still Kohsuke, we should provide contact information.